enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,454 @@
+
+- `UserNamespacesSupport` is the main toggle for allowing user namespaces in Kubernetes. Without it, the kubelet and kube-apiserver will filter the `hostUsers` field from a pod spec.
+- `UserNamespacesPodSecurityStandards` is for toggling the relaxation of Pod Security Standards when a pod is in a user namespace. This feature will remain in alpha, until the oldest supported kubelet
+  of the newest kube-apiserver supports denying a pod when the feature is enabled but the kernel doesn't support user namespaces.
+- `ProcMountType` is a feature for allowing an `Unmasked` `/proc` mount in a container. While not directly related to user namespaces on the surface, the feature relies on the presence of a user namespace.
+  When paired with a user namespace, the feature opens the opportunity for a container to mount `/proc` mounts internally, which is useful for running nested containers.
+
+`UserNamespacesSupport` will be on by default OOTB in 4.17, as it's beta in 1.30. It is the main featuregate of this feature.
+Enabling `UserNamespacesPodSecurityStandards` and `ProcMountType` will move a cluster into TechPreviewNoUpgrade in 4.17.
+
+`UserNamespacesPodSecurityStandards` is an auxilliary feature gate that loosens Pod Security Standards (PSS) for pods with user namespaces.
+At the time of writing, this relaxation is restricted to the runAsUser/runAsGroup/runAsNonRoot fields.
+However, there are [other](https://github.com/kubernetes/kubernetes/pull/126163) relaxations proposed.
+Since this feature gate is responsible for relaxing validation done on pods that have user namespaces, it's path to GA needs to be done more carefully.
+For instance, Kubernetes supports n-3 skew between the kube-apiserver and kubelet. If the kube-apiserver supports user namespaces, but the kubelet doesn't,
+then the kube-apiserver may expect a pod is confined by a user namespace when it is not, and accidentally give it more privileges than it deserves.
+Thus, `UserNamespacesPodSecurityStandards` is currently in alpha, and will not advance to GA until three releases after Kubelet supports the `UserNamespacesSupport`
+feature gate.
+This feature gate is also the main feature gate used to toggle the SCC changes above. This is done because the same version skew issue exists in Openshift,
+and it would be better to keep UserNamespacesSupport on by default (not skewing from upstrem).
+
+ProcMountType is not directly related to user namespaces, but it does rely on them, and is useful for podman in Openshift pod use cases. Specifically,
+it's needed to allow podman to configure networking for the sub containers (as mounting a sub '/proc' requires the whole '/proc' be read-write, and '/proc' needs
+to be writable to configure sysctls)
+
+The goal is to graduate this feature in 4.18, meaning `UserNamespacesPodSecurityStandards` and `ProcMountType` will be enabled by default, and will not mark
+a cluster as TechPreviewNoUpgrade.
+
+#### MinimalKubeletVersion in the apiserver
```

Comment:
please add the table to this section.  It makes the situation very clear at a glance.

Reply:
added